main.py

```diff
@@ -18,56 +18,113 @@
-from core.controllers import subscriptions
-from core.controllers import subtopic_viewer
-from core.controllers import suggestion
-from core.controllers import topic_editor
-from core.controllers import topic_viewer
-from core.controllers import topics_and_skills_dashboard
-from core.controllers import voice_artist
-from core.domain import user_services
-from core.platform import models
-import feconf
-
-from mapreduce import main as mapreduce_main
-from mapreduce import parameters as mapreduce_parameters
-import webapp2
-from webapp2_extras import routes
+
+ROOT_PATH = os.path.dirname(__file__)
+_PARENT_DIR = os.path.abspath(os.path.join(os.getcwd(), os.pardir))
+OPPIA_TOOLS_PATH = os.path.join(_PARENT_DIR, 'oppia_tools')
+
+# oppia_tools/ is available locally (in both dev and prod mode). However,
+# on the GAE production server, oppia_tools/ is not available, and the default
+# PIL third-party library is used instead.
+#
+# We cannot special-case this using DEV_MODE because it is possible to run
+# Oppia in production mode locally, where a built-in PIL won't be available.
+# Hence the check for oppia_tools instead.
+if os.path.isdir(OPPIA_TOOLS_PATH):
+    PIL_PATH = os.path.join(
+        OPPIA_TOOLS_PATH, 'Pillow-6.2.2')
+    if not os.path.isdir(PIL_PATH):
+        raise Exception('Invalid path for oppia_tools library: %s' % PIL_PATH)
+    sys.path.insert(0, PIL_PATH)
+
+THIRD_PARTY_LIBS = [
+    os.path.join(
+        ROOT_PATH, 'third_party', 'backports.functools_lru_cache-1.6.1'),
+    os.path.join(ROOT_PATH, 'third_party', 'beautifulsoup4-4.9.1'),
+    os.path.join(ROOT_PATH, 'third_party', 'bleach-3.1.5'),
+    os.path.join(ROOT_PATH, 'third_party', 'callbacks-0.3.0'),
+    os.path.join(ROOT_PATH, 'third_party', 'future-0.17.1'),
+    os.path.join(ROOT_PATH, 'third_party', 'gae-cloud-storage-1.9.22.1'),
+    os.path.join(ROOT_PATH, 'third_party', 'gae-mapreduce-1.9.22.0'),
+    os.path.join(ROOT_PATH, 'third_party', 'gae-pipeline-1.9.22.1'),
+    os.path.join(ROOT_PATH, 'third_party', 'graphy-1.0.0'),
+    os.path.join(ROOT_PATH, 'third_party', 'html5lib-python-1.1'),
+    os.path.join(ROOT_PATH, 'third_party', 'mutagen-1.43.0'),
+    os.path.join(ROOT_PATH, 'third_party', 'packaging-20.4'),
+    os.path.join(ROOT_PATH, 'third_party', 'pylatexenc-2.6'),
+    os.path.join(ROOT_PATH, 'third_party', 'simplejson-3.17.0'),
+    os.path.join(ROOT_PATH, 'third_party', 'six-1.15.0'),
+    os.path.join(ROOT_PATH, 'third_party', 'soupsieve-1.9.5'),
+    os.path.join(ROOT_PATH, 'third_party', 'webencodings-0.5.1'),
+]
+
+for lib_path in THIRD_PARTY_LIBS:
+    if not os.path.isdir(lib_path):
+        raise Exception('Invalid path for third_party library: %s' % lib_path)
+    sys.path.insert(0, lib_path)
+# The system path insertions above MUST be run before the imports below because
+# each of the imports below uses various third_party libraries in their
+# execution. During the import of a pythob module, the python interpreter also
```

Comment:
Spelling: python. Please proofread your comments.

Reply:
Done, sorry read through it but brain must've autotranslated pythob to python. Will be more careful next time.